awx/main/models/workflow.py

```diff
@@ -682,6 +681,28 @@ def get_ancestor_workflows(self):
             wj = wj.get_workflow_job()
         return ancestors
 
+    def get_effective_artifacts(self, parents_set=None):
```

Comment:
Can we just make this signature `(self, **kwargs)` as well, and get `parents_set` out of the `kwargs` dict?  It will make things just a bit nicer if we wind up adding more things later on.